README.md

```diff
@@ -16,6 +16,39 @@ In particular:
 
 See the documentation for the specific tests for more information and references.
 
+The library is no-std compatible and contains no unsafe code.
+
+Most users will be using the small set of functions exported from the top level, providing "pre-packaged" prime finding functionality with sane defaults.
+
+## Example
+
+Find a 196 bit prime returned in a 256-bit long `crypto_bigint::U256`:
+
+```rust
+use crypto_bigint::U256;
+let prime = crypto_primes::generate_prime::<U256>(196);
+assert!(crypto_primes::is_prime(&prime));
+```
+
+Find a 64 bit safe prime returned in a `crypto_bigint::U1024`:
+
+```rust
+use crypto_bigint::U1024;
+let prime = crypto_primes::generate_safe_prime::<U1024>(64);
+assert!(crypto_primes::is_safe_prime(&prime));
+```
+
+## Advanced
+
+Advanced users can use the [`hazmat`][hazmat-lnk] module in the library to build a custom prime finding solution that best fit their needs, e.g. by picking different Lucas bases or running Miller-Rabin tests with particular bases.
```

Comment:
The link won't work in the README rendered in github, will it? Maybe just use a hard link to `docs.rs` instead?

Reply:
Although this way the link will be incorrect when viewing pre-release versions... Not sure what's the good way of dealing with it, except for just putting it all into `lib.rs`.

Reply:
It'll show the "latest" published (v0.5.0). I think it's ok and that we can probably release 0.6 soon enough that it doesn't matter so much.